keystore/v2/keystore/filesystem/key.go

```diff
@@ -261,23 +261,19 @@ func (r *KeyRing) addKeyData(data api.KeyData, key *asn1.Key) error {
 		}
 	}
 	switch data.Format {
-	case api.ThemisPublicKeyFormat:
+	case api.ThemisKeyPairFormat:
 		if len(data.PublicKey) == 0 {
 			return api.ErrNoKeyData
 		}
 		newData.PublicKey = data.PublicKey
-
-	case api.ThemisKeyPairFormat:
-		if len(data.PublicKey) == 0 || len(data.PrivateKey) == 0 {
-			return api.ErrNoKeyData
+		if len(data.PrivateKey) != 0 {
```

Comment:
if `len(data.PrivateKey) == 0` we do nothing, but if empty public key, we return ErrNoKeyData? Why not same behavior for private key for `ThemisKePairFormat`?

Reply:
> if `len(data.PrivateKey) == 0` we do nothing, but if empty public key, we return ErrNoKeyData? Why not same behavior for private key for ThemisKePairFormat?

Because it makes sense to have a public part of the key pair stored without a corresponding private key. We definitely don’t want to store neither of them. It also makes little sense to allow to store the private key alone, given how we use it with Themis. Themis allows to specify only private key for signature verification in Secure Message but we don’t really use it that way in Acra.